### ISSUE TITLE
Proposal: Install helm plugin from archive

### ISSUE THREAD

**Issue by bastienbosser** [helm/helm#12684]
I'm opening this issue because I'm currently encountering the need to install helm plugins on clusters that don't have internet access. It would therefore be interesting if the helm plugin install command could use an archive as argument.
Currently, to get around the difficulty of installing the helm-mapkubeapis plugin, I'm running this subset of the command.

```
mkdir /root/.local/share/helm/plugins/helm-mapkubeapis
tar xzf helm-mapkubeapis_0.4.1_linux_amd64.tar.gz --directory /root/.local/share/helm/plugins/helm-mapkubeapis/
mkdir /root/.local/share/helm/plugins/helm-mapkubeapis/bin
mv /root/.local/share/helm/plugins/helm-mapkubeapis/mapkubeapis /root/.local/share/helm/plugins/helm-mapkubeapis/bin/
```

**Comment by gjenkins8** [comment]
Can you describe in more detail the problem please

```
$ helm plugin install --help

This command allows you to install a plugin from a url to a VCS repo or a local path.

Usage:
  helm plugin install [options] <path|url>... [flags]
 ```
  
 It should be possible to install a plugin from a path. Do you mean the path could be an archive? Rather than an 
 (unpacked) directory?

**Comment by bastienbosser** [comment]
Yes, I'd like to be able to install a plugin via a path that happens to be an archive. Since plugin can be downloaded directly as archives, it makes more sense to use a path which is an archive. I am opening this proposal following a number of difficulties encountered in my attempt to install the helm-mapkubeapis plugin with a path. I've realized that the script that is executed to perform the install is not compliant with this method. As this plugin is part of the helm projects, I have opened this issue in order to make a functional improvement proposal, but also to better understand whether the guidelines that are proposed in the development of plugin are respected for this one.
